### PR TITLE
add no-require-master flag to doctr deploy

### PR DESCRIPTION
**Pull request by gforsyth**

### CODE REVIEW

File: doctr/travis.py [hunk 1]
```diff
@@ -150,8 +150,9 @@ def setup_GitHub_push(deploy_repo, auth_type='deploy_key', full_key_path='github
     TRAVIS_BRANCH = os.environ.get("TRAVIS_BRANCH", "")
     TRAVIS_PULL_REQUEST = os.environ.get("TRAVIS_PULL_REQUEST", "")
 
-    if TRAVIS_BRANCH != "master":
-        print("The docs are only pushed to gh-pages from master", file=sys.stderr)
+    if TRAVIS_BRANCH != "master" and no_require_master:
+        print("The docs are only pushed to gh-pages from master. To force pushing from "
```
Comment (asmeurer) [inline doctr/travis.py]: "To *allow* pushing". Force pushing sounds like `git push --force`, which we are definitely not doing.

File: doctr/__main__.py [hunk 2]
```diff
@@ -64,6 +64,8 @@ def get_parser():
         help="""Temporary directory used on gh-pages. The default is %(default)r.""")
     deploy_parser.add_argument('--deploy-repo', default=None, help="""Repo to
         deploy the docs to. By default, it deploys to the repo Doctr is run from.""")
+    deploy_parser.add_argument('--no-require-master', dest='nomaster', action='store_true',
```
Comment (asmeurer) [inline doctr/__main__.py]: I find it easier to invert the `--no` options (see `--no-upload-key` for example).

File: doctr/travis.py [hunk 3]
```diff
@@ -132,7 +132,7 @@ def get_current_repo():
     _, org, git_repo = remote_url.rsplit('.git', 1)[0].rsplit('/', 2)
     return (org + '/' + git_repo)
 
-def setup_GitHub_push(deploy_repo, auth_type='deploy_key', full_key_path='github_deploy_key.enc'):
+def setup_GitHub_push(deploy_repo, auth_type='deploy_key', full_key_path='github_deploy_key.enc', no_require_master=False):
```
Comment (asmeurer) [inline doctr/travis.py]: Should be `require_master=True`.

File: doctr/travis.py (hunk 1 above)
Comment (gforsyth) [inline doctr/travis.py]: Yep, that's a good distinction to make.

File: doctr/__main__.py (hunk 2 above)
Comment (gforsyth) [inline doctr/__main__.py]: got it, changes pushed up

File: doctr/__main__.py [hunk 4]
```diff
@@ -105,7 +107,8 @@ def deploy(args, parser):
     deploy_repo = args.deploy_repo or build_repo
 
     if setup_GitHub_push(deploy_repo, auth_type='token' if args.token else
-        'deploy_key', full_key_path=args.key_path):
+                         'deploy_key', full_key_path=args.key_path,
+                         no_require_master=args.nomaster):
```
Comment (asmeurer) [inline doctr/__main__.py]: `require_master=args.require_master`

**Comment by asmeurer** [comment]
+1 once you fix the error.

**Pull request merged**